x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrail.java

```diff
@@ -72,4 +73,7 @@ void runAsDenied(String requestId, Authentication authentication, String action,
     void runAsDenied(String requestId, Authentication authentication, RestRequest request,
                      AuthorizationInfo authorizationInfo);
 
+    void explicitAccessEvent(String requestId, AuditLevel eventType, Authentication authentication, String action, String[] indices,
```

Comment:
Why do we need this method? The name isn't conveying a meaning to me

Reply:
Indeed an explanation is due.
This PR does two things:
* removes duplicated index (alias) names for `BulkShardRequests`
* audits only the resolved index (alias) name for each `BulkItemRequest` instead of the complete indices (aliases) set in the containing `BulkShardRequests`.

This method is required for the latter. We normally audit `TransportMessage` actions, but `BulkItemRequest` does not extend that.
I tried another path, to have a dedicated audit method for `BulkShardRequests` (which extends `TransportMessage`) that would audit the individual `BulkItemRequests` components in one pass. But given the way we authorize bulk items **after** the original request was authorized and audited, the refactoring would have to move the `AuthorizationResultListener` after we handle bulk items in `handleIndexActionAuthorizationResult`. This would violate the "norm" that `authorizeIndexAction`s listener is always an `AuthorizationResultListener` and would increase the complexity because the original bulk shard could be unauthorized and we should audit this and abort, and not go through authorizing comprising items and audit those.

I would like to try to solve this another time, unless you feel otherwise.

Reply:
Can we make the method name `indexAccessEvent` or `explicitIndexAccessEvent` and add javadocs to include some of the detail from your comment?

Reply:
I have renamed it `explicitIndexAccessEvent`. There was a javadoc at the implementation in `LoggingAuditTrail`, I have moved it here.